src/solvers/z3/z3_conv.cpp

```diff
@@ -36,6 +38,8 @@ z3_convt::z3_convt(const namespacet &_ns, const optionst &_options)
   p.set("relevancy", 0U);
   p.set("model", true);
   p.set("proof", false);
+  if(options.get_bool_option("z3-debug"))
+    p.set("smtlib2_log", "log.smt2");
```

Comment:
Shall we allow the user to provide the filename for the log?

Reply:
done